postgresql/client.go

```diff
@@ -61,42 +60,40 @@ func ParseFlags(cfg *Config) *Config {
 
 // Client sends Prometheus samples to PostgreSQL
 type Client struct {
-	logger log.Logger
-	db     *sql.DB
-	cfg    *Config
+	db  *sql.DB
+	cfg *Config
 }
 
 // NewClient creates a new PostgreSQL client
-func NewClient(logger log.Logger, cfg *Config) *Client {
+func NewClient(cfg *Config) *Client {
 	connStr := fmt.Sprintf("host=%v port=%v user=%v dbname=%v password='%v' sslmode=%v connect_timeout=10",
 		cfg.host, cfg.port, cfg.user, cfg.database, cfg.password, cfg.sslMode)
 
-	if logger == nil {
-		logger = log.NewNopLogger()
-	}
+	wrappedDb, err := util.RetryWithFixedBackoff(10, time.Second, func() (interface{}, error) {
```

Comment:
Make the num retries and delay configurable. E.g., if one sets 0 retries (maybe default) than the behavior is like before.

Generally, I think apps should fast fail if not otherwise specified. Otherwise it might look like the app is running just fine when it actually isn't.